tls/s2n_client_hello.c

```diff
@@ -672,15 +678,13 @@ int s2n_client_hello_recv(struct s2n_connection *conn)
         conn->client_hello.callback_invoked = true;
 
         /* Call client_hello_cb if exists, letting application to modify s2n_connection or swap s2n_config */
-        if (conn->config->client_hello_cb) {
+        if (conn->config && conn->config->client_hello_cb) {
             int rc = conn->config->client_hello_cb(conn, conn->config->client_hello_cb_ctx);
             POSIX_GUARD_RESULT(s2n_client_hello_process_cb_response(conn, rc));
         }
     }
 
-    if (conn->client_hello_version != S2N_SSLv2) {
-        POSIX_GUARD(s2n_process_client_hello(conn));
-    }
+    POSIX_GUARD(s2n_process_client_hello(conn));
 
```

Comment:
There's a lot more processing in s2n_process_client_hello than the snippets you removed from s2n_sslv2_client_hello_parse. How are you confident that this doesn't change sslv2 behavior?

Reply:
I'm thinking it might be better to not include the SSLV2 changes. Given that we have no integration test for sslv2, I really have no confidence that our code is currently working or will continue working after this refactor. It does mean that there is still one place where we reference the config before the callback, but I doubt this code is frequently executed anyways.